daemon/src/projection.rs

```diff
@@ -828,26 +829,70 @@ impl Tx {
         let _ = self.quote.btc_usd.send(quote.map(|q| q.into()));
     }
 
-    fn send_order_update(&self, symbol: ContractSymbol, offer: MakerOffers) {
-        match symbol {
-            ContractSymbol::BtcUsd => {
-                self.order.btc_usd.send(offer).unwrap_or_default();
-            }
-            ContractSymbol::EthUsd => {
-                self.order.eth_usd.send(offer).unwrap_or_default();
-            }
-        }
+    fn send_offer_update(&self, offers: &Offers) -> Result<()> {
+        let OffersByContractSymbol { btc_usd, eth_usd } = split_offers_by_contract_symbol(offers);
+
+        self.order.btc_usd.send(btc_usd)?;
+        self.order.eth_usd.send(eth_usd)?;
+
+        Ok(())
     }
 }
 
 /// Internal struct to keep state in one place
 struct State {
     network: Network,
     quote: Option<xtra_bitmex_price_feed::Quote>,
+    offers: Offers,
```

Comment:
❓ Keeping the state for the "current / latest offers" up to date in multiple places in the codebase can be error prone. We already have the `latest_offers` in multiple places.  
What is the motivation for this (avoid one more message between actors...?)? 
I do think it makes sense to push it and store it, but I wonder if this could lead to unwanted side-effects where the state is not up to date in some cases.

Reply:
> What is the motivation for this (avoid one more message between actors...?)?

No messages have changed. The `projection::Actor` still hears about the most recent offers via the `Update` message. What has had to change is the fact that we no longer necessarily send _all_ the current offers from maker to taker every time we run the `/itchysats/offer`. It's perfectly acceptable for the maker to only tell us about ETHUSD offers if that's the only thing that has changed.

With that in mind, I thought that it would be useful to keep a holistic view of the offers in the `projection::Actor` and update it every time we receive an `Update` message. Thinking about it now I am not totally sure that it is necessary. I wonder if it's always sufficient to relay the offers that have changed via the feed. What do you think? Keep in mind that this is different to what we used to do. Before we would always receive _all_ the offers (all BTCUSD) and pump that into the feed.

> I do think it makes sense to push it and store it, but I wonder if this could lead to unwanted side-effects where the state is not up to date in some cases.

With the current functionality in this PR what we are doing is:

1. Receive an update with the new offers.
2. Combine our state with the new offers.
3. Pushing all offers to the feeds.

I think maybe we just need:

1. Receive an update with the new offers.
2. Relay the new offers no matter what.

It's probably bad to send events for offers that haven't actually changed! I'm not 100% sure about how we're supposed to use SSE though, so maybe you have more thoughts on this.

Reply:
I'd say it's better to tackle the state of the `Offer` throughout the application (and multiple actors) in a separate follow-up. You can decide if you want to keep it in the projection as you implemented. 
Tbh I don't think it will make a big difference, given that offers are regularly published anyway, resulting in new `OfferId`s which in turn will require an update.